### PR TITLE
Feature/accurate respond to

### PR DESCRIPTION
**Pull request by somebody32**
#### :tophat: What? Why?

Review it after the [previous one](https://github.com/teambox/choices/pull/2)

This pr adds accurate `respond_to?`, before that was missing this check `@choices.key?(method)`

### CODE REVIEW

File: lib/choices.rb [hunk 1]
```diff
@@ -26,12 +26,12 @@ def with_local_settings(filename, env, suffix)
...
     end
   end
-  
+
   def yaml_load(content)
     if defined? YAML::ENGINE
       # avoid using broken Psych in 1.9.2
       old_yamler = YAML::ENGINE.yamler
-      YAML::ENGINE.yamler = 'syck'
+      YAML::ENGINE.yamler = 'psych'
```
Comment (saimonmoore) [inline lib/choices.rb]: this used to be a problem but I take it you've tested this thoroughly...

Comment (saimonmoore) [inline lib/choices.rb]: it's likely it's not an issue anymore

**Comment by saimonmoore** [comment]
:postal_horn: && :+1:

**Comment by somebody32** [comment]
@saimonmoore rebased and cleaned

**Comment by csaura** [comment]
:+1:

**Pull request merged**